src/middleware/auth.js

```
@@ -0,0 +1,43 @@
+// src/middleware/auth.js
+const { verifyAccessToken, rotateRefreshToken, issueAccessToken } = require('../services/tokenService');
+
+/**
+ * Auth middleware:
+ * - Tenta validar o access token (header Bearer ou cookie)
+ * - Se inválido/ausente, tenta refresh cookie no DB:
+ *    - Se ok, rotaciona refresh, emite novo access, setta cookies e continua
+ *    - Se falhar, 401
+ */
+module.exports = async function auth(req, res, next) {
+  try {
+    const bearer = req.header('Authorization')?.replace('Bearer ', '');
+    const cookieAT = req.cookies?.accessToken;
+    const token = bearer || cookieAT;
+
+    if (token) {
+      const payload = verifyAccessToken(token);
+      req.user = { id: payload.sub };
+      return next();
```

The top-level comment promises a fallback to refresh when the access token is invalid *or* absent. Currently the code only falls back if there is no access token; if `verifyAccessToken` throws (expired/invalid token) the exception is caught by the outer try/catch and returns 401 without attempting refresh. Change the flow to attempt verification and, on verification failure, proceed to the refresh-token fallback instead of returning immediately. This restores the intended behavior described in the file header (try refresh when access is invalid).